server/src/main/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhase.java

```diff
@@ -537,11 +537,24 @@ private static List<SearchShardIterator> sortShards(
         MinAndMax<?>[] minAndMaxes,
         SortOrder order
     ) {
-        return IntStream.range(0, shardsIts.size())
```

Comment:
This is an unfortunate implementation to begin with, but it's hard to do this on top of an `int[]` I guess. At least doing it this way is quite a bit faster by not instantiating a tree map under the hood (from `sorted()` and having a lot less function calls.